packages/react-sdk/docusaurus/docs/React/02-tutorials/03-livestream.mdx

```diff
@@ -1,6 +1,341 @@
+  const isBroadcasting = useIsCallBroadcastingInProgress();
+  return (
+    <Button onClick={() => {
+        if (isBroadcasting) {
+          call.stopBroadcasting();
+        } else {
+          call.startBroadcasting();
+        }
+    }}>
+      {isBroadcasting ? 'Stop Stream' : 'Start Stream'}
+    </Button>
+  )
+};
+
+```
+
+:::info
+Starting and stopping a live stream for the viewers is an operation that takes a few seconds to complete.
+It is a good practice to show a loading indicator while the operation is in progress.
+
+There are two ways how we can do that:
+- have a `useEffect` that would run whenever the we have initiated a change in the broadcasting state of the call until the value of `isBroadcasting` toggles.
+- attach an event listener to the call object and follow the delivery of `call.broadcasting_started` and `call.broadcasting_stopped` events. Read more about events [here.](../10-advanced/06-events.mdx)
+:::
+
+## Step 4 - Setup the viewer mode
+
+The viewer mode is very similar to the host mode. The only difference is that the viewer doesn't have to publish their audio and video to the live stream.
+
+A viewer can join a live stream in two ways:
+- WebRTC-based live streaming (ultra-low latency), by using Stream Edge servers
+- HLS-based live streaming (slight delay, but better buffering), by using Stream CDN
+
+### WebRTC-based live streaming
+
+For the WebRTC, we are going to use the same `<StreamMeeting />` component as in the host mode. The only difference is that we are going to use an anonymous user for the viewer.
+
+:::caution
+Anonymous users can't establish a WebSocket connection to our backend, therefore they are unable to receive any events from the call.
+This means that the viewer won't know when the call starts or stops broadcasting.
+
+To circumvent this limitation, the Viewer app should be able to be notified by the Host app or some other backend service when the call starts or stops broadcasting.
+Alternatively, you can keep polling the backend for the current state of the call. For more details on this, see [Querying Calls.](../02-guides/06-querying-calls.mdx)
+:::
+
+```tsx
+import {
+  useCall,
+  useCallCallingState,
+  LoadingIndicator,
+  PaginatedGridLayout,
+  StreamMeeting,
+  StreamVideo,
+} from '@stream-io/video-react-sdk';
+
+// ... API key, provider, and user setup
+
+export const ViewerApp = () => {
+  const client = useCreateStreamVideoClient({
+    apiKey,
+    tokenOrProvider,
+    user,
+    isAnonymous: true,
+  });
+
+  const [activeCall, setActiveCall] = useState<Call>();
+  useEffect(() => {
+    if (!callId) return;
+    client
+      .queryCalls({
+        filter_conditions: {
+          cid: { $in: [`livestream:${callId}`] },
+        },
+        sort: [{ field: 'cid', direction: 1 }],
+      })
+      .then(({ calls }) => {
+        const [call] = calls;
+        if (call) {
+          setActiveCall(call);
+        }
+      });
+  }, [client, callId]);
```

Comment:
It is not entirely clear to me, from where the `callId` comes. That could happen also to other readers.

Reply:
I've added more information.